src/java/org/apache/cassandra/utils/Backoff.java

```diff
@@ -21,23 +21,55 @@
 import java.util.concurrent.TimeUnit;
 import java.util.function.DoubleSupplier;
 
+import org.apache.cassandra.config.RetrySpec;
+import org.apache.cassandra.repair.SharedContext;
+import org.apache.cassandra.tcm.Retry;
+
 public interface Backoff
 {
-    /**
-     * @return max attempts allowed, {@code == 0} implies no retries are allowed
-     */
-    int maxAttempts();
-    long computeWaitTime(int retryCount);
+    boolean mayRetry(int attempt);
+    long computeWaitTime(int attempt);
     TimeUnit unit();
 
+    static Backoff fromRetry(Retry retry)
```

Comment:
Should we maybe try and unify TCM `Retry` and this one fully? I do not think they contradict each other, they just use different hierarchies: one has backoff on the top, and retry is just a function of this, and the other has this other way around.

Reply:
would make sense but didn't want to do in this patch.  TCM has 2 usages I saw

1) network - this patch unifies
2) blocking - accord and repair are non-blocking so don't have a reason for stateful (other than jitter) impl

Reply:
Sounds good; if you like - I can take over this one.